public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -32,6 +31,7 @@ import { getTimeSrv, TimeSrv } from 'app/features/dashboard/services/TimeSrv';
 import { getTemplateSrv, TemplateSrv } from 'app/features/templating/template_srv';
 
 import { RowContextOptions } from '../../../features/logs/components/LogRowContextProvider';
+import { getLogLevelFromKey } from '../../../features/logs/utils';
```

Comment:
hmmm, but this would mean that `getLogLevelFromKey` can't be used by external plugins. And it seems that both Loki & Elastic use it. Shouldn't we keep this one in `grafana/data`?

Reply:
good catch!

so, hmm.. this is complicated 😄     : 
1. first i agreed with you, i reverted the change for this, pushed the commit to the PR,  so `getLogLevelFromKey` stays in `grafana-data`
2. but then i looked into it more... in both Loki & Elastic , `getLogLevelFromKey` is used for a single reason: to calculate the logs-volume-histogram. to be more exact, to fill this structure: https://github.com/grafana/grafana/blob/main/public/app/core/logsModel.ts#L666 .. it is used by the `queryLogsVolume` helper function, which is inside grafana-main. so it is not available to any external plugin. so, the reason why loki&elastic uses `getLogLevelFromKey`, that reason does not currently exist for external datasources. so, my opinion currently is to remove it from `grafana-data`, we can add it to some "available for external datasources" place when we make logs-volume-histogram "avabilable for external datsources", because currently logs-volume-histogram API is marked `@internal`.

WDYT?

Reply:
Okay, sounds good. Thanks for looking into it.